third_party/move/move-borrow-graph/src/graph.rs

```diff
@@ -110,6 +110,32 @@ impl<Loc: Copy, Lbl: Clone + Ord> BorrowGraph<Loc, Lbl> {
         returned_edges
     }
 
+    /// Returns true if the given reference is borrowed via the given label.
```

Comment:
What was nice about the previous reference safety algorithm was that it explained all the core representations giving intuitions about them. Would be nice to have something similar here since you already understand this deeply, but it is not a PR blocker.

Reply:
Yes, would need to pep up this code here and document better.